h2/stream.py

```diff
@@ -701,6 +701,22 @@ def __init__(self,
         # The configuration for this stream.
         self.config = config
 
+    def __repr__(self):
+        stream_state_strs = {
+            StreamState.IDLE: "Idle",
+            StreamState.RESERVED_REMOTE: "Reserved Remote",
+            StreamState.RESERVED_LOCAL: "Reserved Local",
+            StreamState.OPEN: "Open",
+            StreamState.HALF_CLOSED_REMOTE: "Half Closed Remote",
+            StreamState.HALF_CLOSED_LOCAL: "Half Closed Local",
+            StreamState.CLOSED: "Closed"
+        }
```

Comment:
Do we need to do anything fancy here? Glancing at [the documentation](https://docs.python.org/3.5/library/enum.html?highlight=intenum) for `IntEnum`, I think we could get away with using the repr of `self.state_machine.state`. For example:

```pycon
>>> from h2.stream import StreamState
>>> repr(StreamState.IDLE)
'<StreamState.IDLE: 0>'
>>> repr(StreamState.RESERVED_LOCAL)
'<StreamState.RESERVED_LOCAL: 2>'
>>> state_machine_state = StreamState.CLOSED
>>> repr(state_machine_state)
'<StreamState.CLOSED: 6>'
```

Admittedly it’s not ideal that we have two sets of angle brackets floating around, but I think using that is simpler and safer than writing a separate lookup table.